[Library] LibraryTitle, Download 컴포넌트 초안

ingbinsee opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- Library 페이지의 기능 및 레이아웃(header, main, nav, footer 등 공통 section 제외)을 구현하였습니다.

## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 분기점(모바일, 태블릿, 데스크탑)에 따른 반응형 레이아웃 구현
- Download 컴포넌트 제작 및 클릭 시 파일 다운로드 기능 구현

https://github.com/ReOpenSG/ReOpenSG/assets/140426866/d1dda91d-2c5b-495e-857f-6ca50281a034

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

- 회사 소개서, 회사 브로슈어 영문 버전 다운로드는 추후 디자인이 확정되는 대로 작업하겠습니다.

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

ingbinsee committed:
✨ [Feat] : Library 페이지 Title 컴포넌트 완료

Title 컴포넌트를 구현하였습니다.
+ 분기점(모바일, 태블릿, 데스크탑)에 따른 변화를 반영하였습니다.

Related to: #31

ingbinsee committed:
✨ [Feat] : Download 컴포넌트 완료

Library 페이지의 회사 소개서, 회사 브로슈어를 다운로드할 수 있는 섹션을 추가하였습니다.
+ 국문, 영문 아이콘 클릭 시 해당하는 파일이 다운로드됩니다.
+ 분기점(모바일, 태블릿, 데스크탑)에 따른 변화를 반영하였습니다.
+ module.css로 스타일을 분리하였습니다.

! 국문, 영문 다운로드 아이콘은 임의로 넣어두었습니다.
! 다운로드 파일을 임의로 assets에 추가해 두었습니다. 추후 Firebase를 활용하여 관리 예정입니다.

Related to: #32

ingbinsee committed:
✨ [Feat] : Download 컴포넌트 재작업

Download 컴포넌트 디자인 미확정에 따라 기존 시안으로 작업 되돌렸습니다.

! 추후 Download 컴포넌트 디자인 확정 시 수정 예정입니다.

Related to: #32

ingbinsee committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ryudg commented on src/components/Library/DownloadCard.jsx:
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DownloadIcon from './DownloadIcon';
+// import DownloadIcon from './DownloadIcon';
```
**P3**: 불필요한 주석 삭제 바랍니다

ingbinsee replied:
불필요한 주석 삭제하였으며, tailwindcss로 작성된 코드 module.css로 분리하였습니다.

dani-day commented on src/components/Library/DownloadCard.jsx:
```diff
@@ -0,0 +1,41 @@
+        <div className="flex flex-col-reverse gap-open-xl">
+          <h4 className="-text--openfoundation-primary font-open-label text-open-font-large tablet:text-open-font-xxl desktop:text-open-font-xxl">
+            <span>
+              OPENSG
+              <br />
+              {text}
+            </span>
```
P1: 시안에서 파일 제목에 OpenSG 가 중복으로 사용되어 오해가 있었던 것 같습니다. OPENSG 라는 단어가 고정으로 있는 것이 아니라 파일명마다 달라지는 부분이었기 때문에 이 부분 수정해주시면 감사하겠습니다 :)

ingbinsee replied:
해당 부분 props로 분리하였습니다.

ryudg reviewed (approved):
**P3**: 이후 className으로 작성되어있는 TailwindCSS를 module.css로 분리하는게 더 좋을것 같습니다.

dani-day commented on src/components/Library/DownloadSection.jsx:
```diff
@@ -0,0 +1,27 @@
+        category="소개서"
+        href={companyProfileKo}
+        download="OPENSG_회사_소개서_국문"
+        ariaLabel="OPENSG 회사 소개서 국문"
```
P3: Voiceover 로 읽어보았을 때 ariaLabel 로 작성된 부분이, O, P, E, N, S, G 로 각 철자를 나열한 후 회사 소개서 국문을 읽고 있습니다. 저희가 읽는 방식과 같은 방식으로 정보가 제공되려면 '오픈에스지 회사 소개서 국문' 으로 한글 작성이 되어야할 것 같습니다. 기능이나 스타일링 문제가 아니라 추가로 확인해본 부분이라 P3 으로 두겠습니다 :)

ingbinsee replied:
미처 확인하지 못한 부분이네요. 감사합니다.
해당 사항 반영하였습니다.

kelvin-chihyun reviewed (approved):
P3: 확인했고, 특이사항 없어보입니다

dani-day commented on src/components/Library/DownloadCard.jsx:
P1: 제가 피그마 시안에 디자인 내용을 더 자세히 작성해두지 않고, 파일이 2개 이하가 됐을 경우를 고려하지 않아 차이가 생긴 것으로 보입니다. 카드의 max-width 를 400 으로 설정해주시고, 대신에 카드 전체 정렬을 space-between 이 아니라 gap 을 gutter 인 36px 로 적용해주시면 감사하겠습니다. 해당 내용을 피그마 시안에도 반영해두었습니다. 피그마에서 카드가 하나 빠졌을 때 어떤 모양이 되는지 확인해보실 수 있을 것 같습니다. 설명이 견고하지 않아 번거롭게 해드렸네요 ㅠ 죄송합니다

ingbinsee replied:
max-width 설정하였고, space-between 대신 gap 16px(모바일), 36px(태블릿, 데스크탑) 설정하였습니다.

dani-day commented on src/components/Library/DownloadSection.jsx:
```diff
@@ -0,0 +1,27 @@
+
+function DownloadSection() {
+  return (
+    <div className="flex flex-wrap justify-center gap-open-gutter-mobile tablet:gap-open-gutter-desktop desktop:gap-open-gutter-desktop grow m-open-xl tablet:m-open-5xl desktop:m-open-5xl">
```
P1: Library.jsx 에 전체 px margin 이 적용되어있으므로, Section 별로 별도 margin 이 적용되어있으면 안될 것 같습니다. 예를들어 375px 모바일 사이즈를 확인한 경우, 피그마 시안과 상이하게 좌우 여백이 더 크게 적용되어있습니다. 이 부분 레이아웃 스타일링 수정이 필요할 것 같습니다. 확인부탁드리겠습니다 :)

ingbinsee replied:
해당 사항 반영하였습니다.

dani-day reviewed (changes requested):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

- DownloadCard.jsx (P1 review 2개)
- DownloadSection.jsx(P1 review 1개, P3 review 1개)

ingbinsee committed:
🌈 [Design] : module.css로 CSS 정리

tailwindcss로 작업된 디자인 코드를 module.css로 변경하였습니다.

ingbinsee committed:
♻️ [Refactor] : 리뷰 사항 반영

+ DownloadCard.jsx에서 props를 분리하였습니다.
+ DownloadSection.jsx에서 Voiceover의 결과를 반영하여 aria-label을 수정하였습니다.
+ DownloadCard.jsx에서 max-width를 설정하였으며, space-between 대신 gap을 설정하였습니다.
+ DownloadSection.jsx에서 설정된 margin 값을 삭제하였습니다.

Related to: #33

dani-day commented on src/components/Library/Library.module.css:
```diff
@@ -0,0 +1,51 @@
+  @apply -text--openfoundation-primary font-open-paragraph text-open-font-small tablet:text-open-font-medium desktop:text-open-font-medium;
+}
+
+.cardWrapper {
```
P2: 피그마에서 라이브러리 반응형 시안 중 desktop 에서는 height 이 500, tablet 에서 400 으로 되어있는 부분이 반영되어있지 않아 보이나 (또는 CSS module 로 분리하면서 누락된 듯 합니다), 추후에 폰트 크기 및 여백 조정하면서 height, width 가 재차 수정될 여지가 있어보여 지금 당장 반영되지는 않아도 될 것 같습니다 :)

dani-day reviewed (approved):
수정 요청드린 항목 반영된 부분 모두 확인하였습니다 👍  
추가 Comment 가 있지만 당장 반영되어야하는 부분은 아닌 것 같아 P2 로 의견 남겼습니다.
